templates/vhost/_ssl.erb

```diff
@@ -16,8 +16,12 @@
   <%- if @ssl_honorcipherorder -%>
   SSLHonorCipherOrder     <%= @ssl_honorcipherorder %>
   <%- end -%>
-  <%- if @ssl_verify_client -%>
+  <%- if @ssl_verify_client and @ssl_verify_client.match('(none|optional|require|optional_no_ca)') -%>
```

Comment:
I think it'd better to have a data type to enforce it. This silently ignores invalid values.

Reply:
Is this a requirement or can be ignored in this case.

Reply:
I think this is a bad pattern and should be avoided. I opened https://github.com/puppetlabs/puppetlabs-apache/pull/2084 which is IMHO the correct pattern. This notifies the user that they did something wrong. With your solution they might find out only in production when it's broken.

Reply:
Thanks @bovy89 for the PR and @ekohl for the suggestion in #2084 - @bovy89 if you incorporate @ekohl 's suggestion to this PR, or want to continue working from there, we'll be happy to merge.

Reply:
done